roomserver/storage/shared/storage.go

```diff
@@ -28,24 +27,48 @@ import (
 const redactionsArePermanent = true
 
 type Database struct {
+	DB *sql.DB
+	EventDatabase
+	Cache              caching.RoomServerCaches
+	Writer             sqlutil.Writer
+	RoomsTable         tables.Rooms
+	StateSnapshotTable tables.StateSnapshot
+	StateBlockTable    tables.StateBlock
+	RoomAliasesTable   tables.RoomAliases
+	InvitesTable       tables.Invites
+	MembershipTable    tables.Membership
+	PublishedTable     tables.Published
+	Purge              tables.Purge
+	GetRoomUpdaterFn   func(ctx context.Context, roomInfo *types.RoomInfo) (*RoomUpdater, error)
+}
+
+// EventDatabase contains all tables needed to work with events
+type EventDatabase struct {
 	DB                  *sql.DB
 	Cache               caching.RoomServerCaches
 	Writer              sqlutil.Writer
 	EventsTable         tables.Events
 	EventJSONTable      tables.EventJSON
 	EventTypesTable     tables.EventTypes
 	EventStateKeysTable tables.EventStateKeys
-	RoomsTable          tables.Rooms
-	StateSnapshotTable  tables.StateSnapshot
-	StateBlockTable     tables.StateBlock
-	RoomAliasesTable    tables.RoomAliases
 	PrevEventsTable     tables.PreviousEvents
-	InvitesTable        tables.Invites
-	MembershipTable     tables.Membership
-	PublishedTable      tables.Published
 	RedactionsTable     tables.Redactions
-	Purge               tables.Purge
-	GetRoomUpdaterFn    func(ctx context.Context, roomInfo *types.RoomInfo) (*RoomUpdater, error)
+}
+
+// GetEventDatabase returns an EventDatabase to work with events only.
+func (d *Database) GetEventDatabase() *EventDatabase {
```

Comment:
Why do we need a getter? Constructors are making the embedded `EventDatabase` anyway, so callers can just do `db.EventDatabase.whatever`?